src/RHEO/Install.sh

```diff
@@ -0,0 +1,103 @@
+# Install/unInstall package files in LAMMPS
+# mode = 0/1/2 for uninstall/install/update
+
+mode=$1
+
+# enforce using portable C locale
+LC_ALL=C
+export LC_ALL
+
+# arg1 = file, arg2 = file it depends on
+
+action () {
+  if (test $mode = 0) then
+    rm -f ../$1
+  elif (! cmp -s $1 ../$1) then
+    if (test -z "$2" || test -e ../$2) then
+      cp $1 ..
+      if (test $mode = 2) then
+        echo "  updating src/$1"
+      fi
+    fi
+  elif (test -n "$2") then
+    if (test ! -e ../$2) then
+      rm -f ../$1
+    fi
+  fi
+}
+
+# package files without dependencies
+action atom_vec_rheo_thermal.h
+action atom_vec_rheo_thermal.cpp
+action atom_vec_rheo.h
+action atom_vec_rheo.cpp
+action compute_rheo_grad.h
+action compute_rheo_grad.cpp
+action compute_rheo_interface.h
+action compute_rheo_interface.cpp
+action compute_rheo_kernel.h
+action compute_rheo_kernel.cpp
+action compute_rheo_rho_sum.h
+action compute_rheo_rho_sum.cpp
+action compute_rheo_surface.h
+action compute_rheo_surface.cpp
+action compute_rheo_vshift.h
+action compute_rheo_vshift.cpp
+action fix_rheo_oxidation.h
+action fix_rheo_oxidation.cpp
+action fix_rheo_pressure.h
+action fix_rheo_pressure.cpp
+action fix_rheo_viscosity.h
+action fix_rheo_viscosity.cpp
+action fix_rheo.h
+action fix_rheo.cpp
+action pair_rheo.h
+action pair_rheo.cpp
+action pair_rheo_solid.h
+action pair_rheo_solid.cpp
+
+# package files with dependencies
```

Comment:
@jtclemm you don't need to do these "soft" dependencies (i.e. install only part of RHEO if BPM is not installed) if you also have a "hard" dependency (exit with an error when BPM is not installed) below.

You need to make a choice, which of the two you want. The hard dependency is obviously much easier to implement and makes some sense here. The soft dependency is more for packages like KSPACE or OPENMP.

If you want to have a soft dependency, then you need to have more CMake scripting to remove the files from RHEO that depend on sources in BPM. That one is rather complex to do (I can do it for you, if you want to go this route). The way I see it, there is little need for the effort (also for future expansions of either BPM or RHEO).

Just let me know your preferences. Since we are after the "stabilization deadline" there is not much of a rush, since this package would only be added after the stable release.

Reply:
That's a good point, I swapped it.